src/renderer/protocol-handler/app-handlers.ts

```diff
@@ -0,0 +1,34 @@
+import { addClusterURL } from "../components/+add-cluster";
+import { clusterSettingsURL } from "../components/+cluster-settings";
+import { extensionsURL } from "../components/+extensions";
+import { landingURL } from "../components/+landing-page";
+import { preferencesURL } from "../components/+preferences";
+import { clusterViewURL } from "../components/cluster-manager/cluster-view.route";
+import { LensProtocolRouterRenderer } from "./router";
+import { navigate } from "../navigation/helpers";
+
+export function bindAppHandlers() {
+  LensProtocolRouterRenderer
+    .getInstance<LensProtocolRouterRenderer>()
+    .addInternalHandler("/preferences", ({ search: { highlight }}) => {
+      navigate(preferencesURL({ fragment: highlight }));
```

Comment:
does `navigate()` already know what to do with the fragment?

Reply:
Here `preferencesURL` (buildURL) has to know what to do with `fragment`, @jim-docker 
`navigate()` just accept string as usual.

Reply:
In this PR I add fragment handling to `buildUrl` so all of the `___URL()` functions can handle fragments.

Reply:
ah, so `"#fred"` in the url corresponds to the id selector to the html element (`<element id=fred />`) on the page (probably basic html navigation that I should know? :) )

Reply:
Yes that is what this means. In #2173 I add an auto-scroll into view and change of colour for the "target" item as well. So that will help catch the eye of the user too.

Reply:
This means we need to be aware that we can't change element id values for exposed "routes"

Reply:
Sort of, we don't define the list of targets anywhere and I could argue that what we do when we detect a fragment matching an `id` as "implementation defined" but that is probably not how we want to go about it.

Reply:
Does this work with #2304? If not -> let's simplify and provide just a route for Preferences page (no fragments).

Reply:
I will test, but I don't see why it shouldn't.

Reply:
@jakolehm it works, will add the scrolling.